### ISSUE TITLE
Only consider MAP_ANNOT_OBLIQUE settings for -RLLX/LLY/URX/URY+r specifications

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/gmt#4859]
**Description of the problem**

*Originally posted in [this comment](https://github.com/GenericMappingTools/gmt/pull/3344#issuecomment-786304660) by @PaulWessel*

The settings for MAP_ANNOT_OBLIQUE should only be considered for rectangular, oblique maps.

**Comment by meghanrjones** [comment]
@PaulWessel, there's a notification above about a commit that closes this issue but it isn't associated with a PR. Is it ready to try out?

**Comment by PaulWessel** [comment]
I started a branch but got stuck in the mud and realized best to just delete it.  So back to square one I am afraid.  It is messy because non-oblique projections are also at times relying on the annon_oblique settings.  Still working on a better approach.

**Comment by PaulWessel** [comment]
Some comments to myself as I consider a working system for a new branch to address this issue.

- Given its name and description, **MAP_ANNOT_OBLIQUE** should really only apply to projections that (a) have an oblique aspect and (b) have straight boundaries.  This is usually mean a combination of **-R -J** that involves **-R..+r**.
- **MAP_ANNOT_OBLIQUE** is internally reduced to a bitflag called `GMT->current.setting.map_annot_oblique`
- However, in addition to that bitflag, _gmt_proj_setup_ will add these assignments for all the miscellaneous projections (Hammer, Sinusoidal etc): `GMT->current.map.frame.horizontal = 2; ` This means "force annotations to be horizontal at the S and N borders".
- The `frame.horizontal `doubles up what **MAP_ANNOT_OBLIQUE** does due to this check at the top of _gmt_map_basemap_:
```
if (GMT->current.setting.map_annot_oblique & GMT_OBL_ANNOT_LON_HORIZONTAL) GMT->current.map.frame.horizontal = 2;
if (GMT->current.map.frame.horizontal == 2) GMT->current.setting.map_annot_oblique |= GMT_OBL_ANNOT_LON_HORIZONTAL;
```
So `current.setting.map_annot_oblique` ends up being changed due to `current.map.frame.horizontal` even though it has nothing to do with being oblique.
- Even projections that are neither oblique nor miscellaneous still obey the **MAP_ANNOT_OBLIQUE** setting, such as seen in  `gmt basemap -R-30/30/40/60 -JL0/50/40/60/15c -Baf -pdf t --MAP_ANNOT_OBLIQUE=lon_horizontal`
- Finally, **subplot** decides that y-annotations should be parallel to the frame and will update _map_annot_oblique_ so that we get this effect:` gmt basemap -R-30/30/0/40 -JM15c -Baf -pdf t --MAP_ANNOT_OBLIQUE=lat_parallel`. So we don't want to loose that capability.

The problem I see here is not so much that the user might want that last effect and GMT can deliver it.  The problem is that if we decide a particular choice of bits for **MAP_ANNOT_OBLIQUE** should become new defaults (e.g., **lon_horizontal,lat_horizontal,tick_extend**), they propagate to all these other projections were we do not want to change the default. Finally, when GMT parses **MAP_ANNOT_OBLIQUE** it does not yet know what projection will be applied, so we cannot enforce some rule at that point.

Maybe this strategy could work:

1. Let **MAP_ANNOT_OBLIQUE** get a new default meant for oblique projections, say **lon_horizontal,lat_horizontal,tick_extend**.
2. When _gmt_proj_setup_ is called, let it (1) remove/reset these flags if they are inappropriate for the selected projection (e.g., Conic, Cylindrical) and it is _not_ a subplot, and then (2) set the `frame.horizontal = 2` if needed by that projection - this will again affect `map_annot_oblique` later one as we see above.
3. I think we may need to distinguish between using a default _map_annot_oblique_ or one that was user-supplied.  If the user supplied a specific combo then we should _not_ reset it.

So pretty messy implementation, unfortunately.  And of course **MAP_ANNOT_ORTHO** is there to only apply to Cartesian axes.  Perhaps it could be helpful to actually rename **MAP_ANNOT_OBLIQUE** to something different, say **MAP_ANNOT_TYPE** so that we at least don't have to explain why it applies to non-oblique projections as well?  Thoughts, @meghanrjones ?

**Comment by meghanrjones** [comment]
Yes, a bit messy. Regarding the parameter name, I do not think it is necessary to rename **MAP_ANNOT_OBLIQUE** because so many of these parameters are **MAP_** parameters that also apply to non-maps so we're already relying on some extrapolation here.

Do you think that this should be considered a bug that needs to be fixed regardless of whether new defaults are set for MAP_ANNOT_OBLIQUE? If not, I think we should review some maps that use oblique projections before taking the effort to change this. If it should be corrected regardless, then this implementation seems reasonable.

**Comment by PaulWessel** [comment]
The only bug I see really is what happened if I change the default - it had all these unintended consequences for other projections.  If the user want to plot horizontal longitude on a conic map then he should be allowed to even though I think it looks ugly.  But it should not just pop up because we changed the OBLIQUE default.  So I think I can tell if OBLIQUE was updated via file or --par=var and do different things internally.
